### PR TITLE
Merge docker-annotations into kubernetes-annotations.

### PR DESCRIPTION
**Pull request by iocanel**

### CODE REVIEW

**Comment by geoand** [comment]
Tests are complaining :(

File: readme.md [hunk 1]
```diff
@@ -679,20 +679,10 @@ It will not generate or customize the docker build in anyway.
...
-```xml
-<dependency>
-  <groupId>io.ap4k</groupId>
-  <artifactId>docker-annotations</artifactId>
-  <version>${project.version}</version>
-</dependency>
-```  
-To enable the hook you need to add the `@EnableDockerBuild` annotation to your main class.    
-Finally, to trigger the hook, you need to pass `-Dap4k.build=true`  as an argument to the build, for example:
+To trigger the hook, you need to pass `-Dap4k.build=true`  as an argument to the build, for example:
```
Comment (geoand) [inline readme.md]: Wouldn't `@EnableDockerBuild` a few lines down need to be removed as well?

File: annotations/kubernetes-annotations/src/main/java/io/ap4k/kubernetes/annotation/KubernetesApplication.java [hunk 2]
```diff
@@ -32,7 +33,7 @@
 @Buildable(builderPackage = "io.ap4k.deps.kubernetes.api.builder")
 @Pojo(name = "KubernetesConfig", relativePath = "../config",
       mutable = true,
-      superClass = ApplicationConfiguration.class,
+      superClass = BaseConfig.class,
```
Comment (geoand) [inline annotations/kubernetes-annotations/src/main/java/io/ap4k/kubernetes/annotation/KubernetesApplication.java]: I don't fully grok this, but are you sure it's correct?

**Comment by geoand** [comment]
Tests still failing unfortunately :(

File: annotations/kubernetes-annotations/src/main/java/io/ap4k/kubernetes/annotation/KubernetesApplication.java (hunk 2 above)
Comment (iocanel) [reply annotations/kubernetes-annotations/src/main/java/io/ap4k/kubernetes/annotation/KubernetesApplication.java]: What troubles you? Maybe the name is not as descriptive as it should?

ApplicationConfiguration is the parent of all top level config classes.
BaseConfig is the base for all kubernetes classes (KubernetesConfig, OpenshiftConfig).

Comment (geoand) [reply annotations/kubernetes-annotations/src/main/java/io/ap4k/kubernetes/annotation/KubernetesApplication.java]: Got it, makes sense now!

**Comment by geoand** [comment]
Tests pass :100:

**Pull request merged**